### PR TITLE
20 new releases

### PR DESCRIPTION
**Pull request by rjthorne**

### CODE REVIEW

File: json/by-sha256/ff/ff5c3a9b6b98bd547ed2f2dba776f3ec0a7b221fb393a9ba5675d0f110b8bfb9.json [hunk 1]
```diff
@@ -0,0 +1,1375 @@
...
+ "install": {
+  "extract": "{base}/old_town/"
+ },
+ "json_version": "2023.11.06-23.20",
+ "notes": [],
+ "sha256": "ff5c3a9b6b98bd547ed2f2dba776f3ec0a7b221fb393a9ba5675d0f110b8bfb9",
+ "tags": [
+  "author=Eternal_Summer",
+  "commandline=-game old_town",
+  "dependency=",
```
Comment (SpiritQuaddicted) [inline json/by-sha256/ff/ff5c3a9b6b98bd547ed2f2dba776f3ec0a7b221fb393a9ba5675d0f110b8bfb9.json:1360]: Tags should always have key and value:

```suggestion
```

File: json/by-sha256/a0/a04aee22a146df6dd97db37932c7c96028307dab4f6f96e15944bd0e5006bd5b.json [hunk 2]
```diff
@@ -0,0 +1,2215 @@
...
+ },
+ "json_version": "2023.11.06-23.20",
+ "notes": [
+  "Requires one of two specific source ports to play (see readme) and a powerful computer/graphics card."
+ ],
+ "sha256": "a04aee22a146df6dd97db37932c7c96028307dab4f6f96e15944bd0e5006bd5b",
+ "tags": [
+  "author=ComfyByTheFire",
+  "commandline=-game immortal",
+  "dependency=",
```
Comment (SpiritQuaddicted) [inline json/by-sha256/a0/a04aee22a146df6dd97db37932c7c96028307dab4f6f96e15944bd0e5006bd5b.json:2200]: Oops, missed this one

```suggestion
```

**Review by SpiritQuaddicted — CHANGES_REQUESTED** [review]
Lots of critique this time, because I really took some time to look at it. Sorry if it is about stuff that I did not comment about earlier in other PRs.

1. Please stop repackaging files! I am sorry for being so blunt, I think I should have said that much earlier and clearer: Quaddicted should host files made by other people. If files are not in a format that fits the Quake Injector or (similarly basic) other tools, then that is bad luck and not our problem! Repackaging is a waste of time and energy, and occasional cause of bad blood. If you still want to do it, the new file definitely should have a new file name that explicitly tells users that it is not the original version. And the original version should be added too.

For complex installation processes, there needs to be a set of instructions in the metadata for programs to interpret, this is the only sane way. What kind of instructions are needed apart from `extract` and how to model them, is something to be figured out when someone has the time and motivation. I think I shared some rough ideas somewhere in the forums but would need to look that up...

In this PR it's luckily just one release (65894a8).

Again, super huge **sorry** for not being more strict about this earlier.

2. Please don't add notes that simply mention dependencies. This gives us redundant information and is hard to maintain. The correct way to specify dependencies is in the tags. How this gets shown on the website is a different thing (and should definitely be improved).

3. Same for "May require a source port with increased limits / BSP2 support.": If it is specific limits, there should be specific tags. If it requires BSP2, that should be a tag. For now I think the   "exceeds_quake_limits=yes" tag is enough. Free-form text is not useful for programmatic interpretation and that's what is needed for good tools to be based on this all.

4. "Requires one of two specific source ports to play (see readme) (...)": In the future we will need to model this by tags. The actual requirement should be some tags. That way this information would be useful for installer/launcher programs.

The tags are really the juice of it all. All useful information should be modeled in tags if possible. They are currently just a sub-set of what will be useful in the future but to get their, they should really be embraced. We can remodel and change details about them later, but they are the only way to get data in a structured, computer-readable way. And that's what allows tools, engines etc to use this DB for good. https://wiki.openstreetmap.org/wiki/Map_features is the inspiration (that other kind of map ;D )

-----
PS: You definitely deserve cake and more for all the help you've given so far already, thank you so so much!

**Comment by rjthorne** [comment]
Github closed this when I synced forks as there was a commit that went in on the main repo. I've tried reverting but think I'll need to rebase again

But general comments:

- Noted on the repacking. It was clear this time.
- The 'notes' were only ever a temporary feature until you got round to coding in auto-generated notes based on tags. There is not a dependency note without a corresponding tag.
- The feedback on the descriptions is fair. But in honesty given the extent of it I'm now wondering if all descriptions should be removed again as most of them are like this. There are a lot of short ones with stuff that can be surmised in tags, extraneous information, or stuff that could be considered subjective (ironically what qualifies as 'subjective' is also subjective). Once you take those out there isn't really much left for the vast majority of releases

**Comment by SpiritQuaddicted** [comment]
Argh, weird! GitHub seems really confused about this PR. Sometimes it shows me a commit, sometimes not. You don't have to sync unless there are conflicts, I think this branch was free of them even after changes in main.

- Repackaging: Again, sorry for not mentioning it earlier! Must have been a lot of work...
- Notes: Oops, I forgot... I'll see that I make nice dependency and limits messages on the pages from the tags!
- Descriptions: Yeah, it's a bad and widespread problem I think :\ I/We've been subject to quite some flak in the past so that I want to make sure to reduce the possibility of that. We could have something like "official_description" if the authors provide that maybe? Not sure how to handle it all tbh. What I really don't want is people getting into editing wars about descriptions.

**Pull request closed without merging**